flexget/plugins/input/torznab.py

```diff
@@ -0,0 +1,285 @@
+            'imdbid': {'name': 'imdb_id', 'type': str},
+            'infohash': {'name': 'torrent_info_hash', 'type': str},
+            'leechers': {'name': 'torrent_leeches', 'type': int},
+            'rageid': {'name': 'tvrage_id', 'type': int},
+            'season': {'name': 'series_season', 'type': int},
+            'seeders': {'name': 'torrent_seeds', 'type': int},
+            'title': {'name': 'series_name', 'type': str},
+            'tmdbid': {'name': 'tmdb_id', 'type': int},
+            'traktid': {'name': 'trakt_id', 'type': int},
+            'tvdbid': {'name': 'tvdb_id', 'type': int},
+            'tvmazeid': {'name': 'tvmaze_series_id', 'type': int},
+            'tvrageid': {'name': 'tvrage_id', 'type': int}
+        }
+        misc = {}
+        for attr in attrs:
+            name = attr.get('name')
+            if name in dictionary.keys():
+                entry[dictionary[name]['name']] = dictionary[name]['type'](attr.get('value'))
+            elif name == 'peers':
+                misc['peers'] = int(attr.get('value'))
+            elif name == 'imdb':
+                misc['imdb'] = str(attr.get('value'))
+            elif name == 'size':
+                misc['size'] = int(attr.get('value'))
+
+        if 'imdb_id' not in entry.keys() and 'imdb' in misc.keys():
+            entry['imdb_id'] = 'tt{}'.format(misc['imdb'])
+
+        if 'peers' in misc.keys():
+            if 'torrent_leeches' not in entry.keys() and 'torrent_seeds' in entry.keys():
+                entry['torrent_leeches'] = misc['peers'] - entry['torrent_seeds']
+            if 'torrent_leeches' in entry.keys() and 'torrent_seeds' not in entry.keys():
+                entry['torrent_seeds'] = misc['peers'] - entry['torrent_leeches']
+
+        if 'content_size' not in entry.keys() and 'size' in misc.keys():
+            entry['content_size'] = old_div(misc['size'], 2 ** 20)
+
+        if 'torrent_seeds' in entry.keys() and 'torrent_leeches' in entry.keys():
+            entry['torrent_availability'] = torrent_availability(
+                entry['torrent_seeds'], entry['torrent_leeches']
+            )
+
+    def _gather_movie_params(self, entry):
+        """Gather query parameters for 'movie' searcher
+
+        Conversion from Flexget Entry fields to znab query parameters
+        https://flexget.com/Entry
+        https://github.com/nZEDb/nZEDb/blob/0.x/docs/newznab_api_specification.txt#L441
+        """
+        params = {}
+        dictionary = {
+            'imdbid': 'imdb_id'
+        }
+
+        for k, v in [item for item in dictionary.items() if item[0] in self.supported_params]:
+            if v in entry.keys() and entry[v]:
+                params[k] = entry[v]
+        for k in ['trakt_movie_name', 'imdb_name', 'movie_name']:
+            if k in entry.keys() and entry[k]:
+                params['q'] = entry[k]
+                break
+
+        return params
+
+    def _gather_tvsearch_params(self, entry):
```

Comment:
This is very similar to `_gather_movie_params`. Maybe it should generalized to take the id lists as input

Reply:
Done